apps/functions/document-check/check-document/__tests__/check-html.test.js

```diff
@@ -0,0 +1,39 @@
+import { validateHTML } from '../validate-html.js';
+
+describe('HTML validation', () => {
```

Comment:
Would be good to validate this against an actual template we have on the live env (the YouTube embed template)

Reply:
I've just tried testing it against [the real template](view-source:https://infrastructure.planninginspectorate.gov.uk/wp-content/ipc/uploads/projects/WW010003/WW010003-001323-ish2%20s1.html), but the HTML on this page contains a lot of comments which are stripped out by `sanitize-html` (there is no option to preserve them). So the validation fails.

I can't see a way around this with the `sanitize-html` library, and annoyingly I can't find a different library that would do what we need (verify that an HTML string matches a predefined template).

Reply:
One option which is a round-about way but might work:

We store the template on the live env as a Handlebars/Nunjucks template with the iframe source parameterised, then
1. An HTML upload comes in
2. Use a regex to extract the iframe src
3. Render the stored template using the extracted src
4. Compare the rendered (correct) result with the actual (possibly invalid) upload HTML